RetailCoder.VBE/Root/RubberduckIoCInstaller.cs

```diff
@@ -0,0 +1,841 @@
+        }
+
+        private static Type[] RubberduckCommandBarItems()
+        {
+            return new Type[]
+            {
+                typeof(ReparseCommandMenuItem),
+                typeof(ShowParserErrorsCommandMenuItem),
+                typeof(ContextSelectionLabelMenuItem),
+                typeof(ContextDescriptionLabelMenuItem),
+                typeof(ReferenceCounterLabelMenuItem),
+#if DEBUG
+                typeof(SerializeDeclarationsCommandMenuItem)
+#endif
+            };
+        }
+
+        private void RegisterParentMenus(IWindsorContainer container)
+        {
+            RegisterParentMenu<UnitTestingParentMenu>(container, UnitTestingMenuItems());
+            RegisterParentMenu<RefactoringsParentMenu>(container, RefactoringsMenuItems());
+            RegisterParentMenu<NavigateParentMenu>(container, NavigateMenuItems());
+            RegisterParentMenu<SmartIndenterParentMenu>(container, SmartIndenterMenuItems());
+            RegisterParentMenu<ToolsParentMenu>(container, ToolsMenuItems());
+        }
+
+        private static void RegisterParentMenu<TParentMenu>(IWindsorContainer container, Type[] menuItemTypes) where TParentMenu : IParentMenuItem
+        {
+            container.Register(Component.For<IMenuItem, TParentMenu>()
+                .ImplementedBy<TParentMenu>()
+                .LifestyleTransient()
+                .DependsOn(Dependency.OnComponentCollection<IEnumerable<IMenuItem>>(menuItemTypes)));
+        }
+
+        private static Type[] UnitTestingMenuItems()
+        {
+            return new Type[]
+            {
+                typeof(RunAllTestsCommandMenuItem),
+                typeof(TestExplorerCommandMenuItem),
+                typeof(AddTestModuleCommandMenuItem),
+                typeof(AddTestMethodCommandMenuItem),
+                typeof(AddTestMethodExpectedErrorCommandMenuItem)
+            };
+        }
+
+        private static Type[] RefactoringsMenuItems()
+        {
+            return new Type[]
+            {
+                typeof(CodePaneRefactorRenameCommandMenuItem),
+#if DEBUG
+                typeof(RefactorExtractMethodCommandMenuItem),
+#endif
+                typeof(RefactorReorderParametersCommandMenuItem),
+                typeof(RefactorRemoveParametersCommandMenuItem),
+                typeof(RefactorIntroduceParameterCommandMenuItem),
+                typeof(RefactorIntroduceFieldCommandMenuItem),
+                typeof(RefactorEncapsulateFieldCommandMenuItem),
+                typeof(RefactorMoveCloserToUsageCommandMenuItem),
+                typeof(RefactorExtractInterfaceCommandMenuItem),
+                typeof(RefactorImplementInterfaceCommandMenuItem)
+            };
+        }
+
+        private static Type[] NavigateMenuItems()
+        {
+            return new Type[]
+            {
+                typeof(CodeExplorerCommandMenuItem),
+                //typeof(RegexSearchReplaceCommandMenuItem),
```

Comment:
can this also be moved into an `#if DEBUG` block?

Reply:
What exactly should be moved there: the `RegexSearchReplaceCommandMenuItem` but uncommented?

Reply:
That would be more in line with the SerializeDeclarationsMenuItem and RefactorExtractMethodCommand bindings. @retailcoder @Hosch250  is the feature ready enough to not blow everything to pieces when we bind it for debug builds?

Reply:
@Vogel612 no idea, @bclothier what do you say?

Reply:
For this PR, I'd leave ExtractMethod out. I'll make a note to update once I can merge.

Reply:
Please note that removing a command menu item from the menus does __not__ cause the corresponding command not to be loaded; it only removes the menu item.